job_creator.php

```diff
@@ -13,23 +13,27 @@ public function getInstallerVersion(string $githubRepository, string $branch): s
         if (in_array($repo, NO_INSTALLER_REPOS)) {
             return '';
         }
-        // e.g. ['4', '11']
-        $portions = explode('.', $branch);
-        if (count($portions) == 1) {
-            return '4.x-dev';
+        // e.g. pulls/4.10/some-bugfix or pulls/4/some-feature
+        if (preg_match('#^pulls/([0-9\.]+)/#', $branch, $matches)) {
+            $branch = $matches[1];
         }
-        if (in_array($repo, LOCKSTEPED_REPOS)) {
-            return '4.' . $portions[1] . '.x-dev';
-        } else {
-            // use the latest minor version of installer
-            $installerVersions = array_keys(INSTALLER_TO_PHP_VERSIONS);
-            // remove '4' version
-            $installerVersions = array_diff($installerVersions, ['4']);
-            // get the minor portions of the verisons e.g. [9, 10, 11]
-            $minorPortions = array_map(fn($portions) => (int) explode('.', $portions)[1], $installerVersions);
-            sort($minorPortions);
-            return '4.' . $minorPortions[count($minorPortions) - 1] . '.x-dev';
+        if (in_array($repo, LOCKSTEPED_REPOS) && is_numeric($branch)) {
+            // e.g. ['4', '11']
+            $portions = explode('.', $branch);
+            if (count($portions) == 1) {
+                return '4.x-dev';
+            } else {
+                return '4.' . $portions[1] . '.x-dev';
+            }
         }
+        // use the latest minor version of installer
```

Comment:
The only situation where this isn't ideal is prefer-lowest won't actually be testing the lowest version of framework according to the constraint in composer.json.... but we can probably live with that.